### PR TITLE
Fix issue because of which the idSchema gets corrupted if the schema has a property by the name "id"

### PR DESCRIPTION
**Pull request by amarnus**

### CODE REVIEW

File: src/components/fields/ArrayField.js [hunk 1]
```diff
@@ -22,15 +22,15 @@ function ArrayFieldTitle({TitleField, idSchema, title, required}) {
   if (!title) {
     return null;
   }
-  const id = `${idSchema.id}__title`;
+  const id = `${idSchema["$id"]}__title`;
```
Comment (n1k0) [inline src/components/fields/ArrayField.js]: You can write `idSchema.$id`

File: test/utils_test.js [hunk 2]
```diff
@@ -511,20 +511,59 @@ describe("utils", () => {
...
+              age: {
+                type: "integer"
+              }
+            },
+            required: [ "id", "age", "name" ]
+          }
+        }
+      };
+      expect(toIdSchema(schema)).eql({
+        "$id": "root",
```
Comment (n1k0) [inline test/utils_test.js]: `$id: "root"` works

File: test/utils_test.js [hunk 3]
```diff
@@ -511,20 +511,59 @@ describe("utils", () => {
...
+                type: "string",
+                enum: [ "M", "F", "Other" ]
+              },
+              age: {
+                type: "integer"
+              }
+            },
+            required: [ "id", "age", "name" ]
+          }
+        }
```
Comment (n1k0) [inline test/utils_test.js]: I don't think we need such a large sample schema; a simple schema like

```js
const schema = {
  type: "object",
  properties: {
    id: {type: "string"}
  }
};
```

is good enough.

**Comment by n1k0** [comment]
Good catch, thanks.

**Comment by amarnus** [comment]
Noted comments. Will update the request with these changes.

**Comment by amarnus** [comment]
Added recommended changes, squashed commits into one and updated request.

**Comment by n1k0** [comment]
Looks good, thanks for contributing.

**Comment by n1k0** [comment]
Released in [v0.36.0](https://github.com/mozilla-services/react-jsonschema-form/releases/tag/v0.36.0).

**Pull request merged**